dpki/src/keypair.rs

```diff
@@ -174,6 +174,33 @@ impl EncryptingKeyPair {
     }
 
     // TODO: Encrypt and decrypt functions
+
+    /// encrypt some arbitrary data with the signing private key
+    /// @param {SecBuf} data - the data to sign
+    /// @return {SecBuf} signature - Empty SecBuf to be filled with the signature
+    pub fn encrypt(&mut self, data: &mut SecBuf) -> HcResult<SecBuf> {
+        let mut nonce = SecBuf::with_insecure(lib3h_sodium::aead::NONCEBYTES);
```

Comment:
It is not secure to encrypt two different messages with the same nonce.

We've got two options.

1 - If we want to keep the simple API, we could randomize the nonce here, and increase the cipher length by `NONCEBYTES`, appending it to the output, so it can be read / used by the decrypt function

or

2 - We could promote nonce to a parameter in both encrypt and decrypt so that the implementor can be put in charge of doing the randomizing / tracking.

(or we could provide both apis so people can choose which one to use?)

Reply:
I like the idea of coming back and implementing with nonce because I think eric and I talked about this and decided it wasn't really needed at this point. But definitely need nonce to be more secure

Reply:
Yep, lets just add in the randomization for now, and then open up the API later.

Reply:
+1 for random nonce now and extra parameter later.